test/security/testrunners/filepermissiontestrunner.go

```diff
@@ -0,0 +1,108 @@
+package testrunners
+
+import (
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/common"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/rule"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/test_runner"
+	"log"
+	"strings"
+	"time"
+)
+
+type FilePermissionTestRunner struct {
+	test_runner.BaseTestRunner
+}
+
+var _ test_runner.ITestRunner = (*FilePermissionTestRunner)(nil)
+
+const agentConfigPath = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
+const agentConfigOnlyRootRead = "-rw-r--r-- 1 root root "
+
+var (
+	onlyRootReadExactMatchRule = rule.Rule{
+		Conditions: []rule.ICondition{
+			&rule.ExactMatch{ExpectedValue: agentConfigOnlyRootRead},
+		},
+	}
+)
+
+var testCases = map[string]rule.Rule{
+	agentConfigPath: onlyRootReadExactMatchRule,
+}
+
+var actualPermissions = make(map[string]string)
```

Comment:
Why does the file no longer exist? This is why I'm trying to understand what we're trying to validate. 

```
[ec2-user@ip-172-31-27-93 bin]$ ls -alh /opt/aws/amazon-cloudwatch-agent/bin | grep config.json
-rwxrwxrwx 1 ec2-user ec2-user 1.5K Jan 30 20:13 config.json
[ec2-user@ip-172-31-27-93 bin]$ sudo /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a fetch-config -s -c file:/opt/aws/amazon-cloudwatch-agent/bin/config.json
****** processing amazon-cloudwatch-agent ******
/opt/aws/amazon-cloudwatch-agent/bin/config-downloader --output-dir /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d --download-source file:/opt/aws/amazon-cloudwatch-agent/bin/config.json --mode ec2 --config /opt/aws/amazon-cloudwatch-agent/etc/common-config.toml --multi-config default
I! Trying to detect region from ec2
D! [EC2] Found active network interface
Successfully fetched the config and saved in /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d/file_config.json.tmp
Start configuration validation...
/opt/aws/amazon-cloudwatch-agent/bin/config-translator --input /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.json --input-dir /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d --output /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.toml --mode ec2 --config /opt/aws/amazon-cloudwatch-agent/etc/common-config.toml --multi-config default
2023/01/30 20:17:32 Reading json config file path: /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d/file_config.json.tmp ...
2023/01/30 20:17:32 I! Valid Json input schema.
I! Detecting run_as_user...
I! Trying to detect region from ec2
D! [EC2] Found active network interface
No csm configuration found.
Configuration validation first phase succeeded
/opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent -schematest -config /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.toml
Configuration validation second phase succeeded
Configuration validation succeeded
Redirecting to /bin/systemctl stop amazon-cloudwatch-agent.service
Redirecting to /bin/systemctl restart amazon-cloudwatch-agent.service
[ec2-user@ip-172-31-27-93 bin]$ ls -alh /opt/aws/amazon-cloudwatch-agent/bin | grep config.json
-rwxrwxrwx 1 ec2-user ec2-user 1.5K Jan 30 20:13 config.json
[ec2-user@ip-172-31-27-93 bin]$ ls -alh /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d
total 4.0K
drwxr-xr-x 2 cwagent cwagent   30 Jan 30 20:17 .
drwxr-xr-x 3 cwagent cwagent  147 Jan 24 16:08 ..
-rw-r--r-- 1 cwagent cwagent 1.5K Jan 30 20:17 file_config.json
[ec2-user@ip-172-31-27-93 bin]$ ls -alh /opt/aws/amazon-cloudwatch-agent/etc/
total 16K
drwxr-xr-x 3 cwagent cwagent  147 Jan 24 16:08 .
drwxr-xr-x 8 root    root     170 Jan 17 16:46 ..
drwxr-xr-x 2 cwagent cwagent   30 Jan 30 20:17 amazon-cloudwatch-agent.d
-rw-rw-r-- 1 cwagent cwagent 2.6K Jan 30 20:17 amazon-cloudwatch-agent.toml
-rw-r--r-- 1 cwagent cwagent  925 Jul 22  2022 common-config.toml
-rw-r--r-- 1 cwagent cwagent   33 Jan 30 20:17 env-config.json
-rw-r--r-- 1 cwagent cwagent   88 Jan 30 20:17 log-config.json
[ec2-user@ip-172-31-27-93 bin]$ sudo /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a stop
****** processing cwagent-otel-collector ******
cwagent-otel-collector has already been stopped

****** processing amazon-cloudwatch-agent ******
Redirecting to /bin/systemctl stop amazon-cloudwatch-agent.service
[ec2-user@ip-172-31-27-93 bin]$ ls -alh /opt/aws/amazon-cloudwatch-agent/etc/
total 16K
drwxr-xr-x 3 cwagent cwagent  147 Jan 24 16:08 .
drwxr-xr-x 8 root    root     170 Jan 17 16:46 ..
drwxr-xr-x 2 cwagent cwagent   30 Jan 30 20:17 amazon-cloudwatch-agent.d
-rw-rw-r-- 1 cwagent cwagent 2.6K Jan 30 20:17 amazon-cloudwatch-agent.toml
-rw-r--r-- 1 cwagent cwagent  925 Jul 22  2022 common-config.toml
-rw-r--r-- 1 cwagent cwagent   33 Jan 30 20:17 env-config.json
-rw-r--r-- 1 cwagent cwagent   88 Jan 30 20:17 log-config.json
[ec2-user@ip-172-31-27-93 bin]$ ls -alh /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d
total 4.0K
drwxr-xr-x 2 cwagent cwagent   30 Jan 30 20:17 .
drwxr-xr-x 3 cwagent cwagent  147 Jan 24 16:08 ..
-rw-r--r-- 1 cwagent cwagent 1.5K Jan 30 20:17 file_config.json
```

Reply:
I don't know. It looks like this log restarted agent before checking that directory
`restart amazon-cloudwatch-agent.service`
Every time I tried to validate post agent stop it failed because of file not found. Only worked after doing this before agent stops. Do you want me to try again?

Reply:
changed some things to address this